lib/Gocdb_Services/Validate.php

```diff
@@ -36,6 +36,22 @@ public function validate($Object_Name, $Field_Name, $Field_Value)
         //Check the length of the field value, if too long, throws exception
         $this->checkFieldLength($Object_Name, $Field_Name, $Field_Value);
 
+        //Check the Type of the field value, where this is defined
+        $type = $this->Get_Field_value($Object_Name, $Field_Name, 'ftype');
+        if (count($type) != 0) {
+          if (strtolower($type) == "string") {
+            if (!is_string($Field_Value)) {
+                throw new \Exception($Field_Name . " must be a string.");
+            }
+          } elseif (strtolower($type) == "boolean") {
+            if (!is_bool($Field_Value)) {
+                throw new \Exception($Field_Name . " must be a boolean.");
```

Comment:
Should we be more specific about the accepted values? Different languages use different things as bools.

Reply:
I am sorry, I don't understand your question.  Which bit should be more specific?

Reply:
Sorry. In the exception message.

Reply:
The exception could come about because a developer has failed to ensure the correct type of thing has made it to this function, in which case the developer knows it is Php. It could occur due to validation in the write API, in which case the nature of what a boolean is dependent on the input language being used (currently only JSON is supported). In the case of the write API, the exception is caught and a link to the documentation is appended. This documentation makes clear what is expected for booleans.

Reply:
Ok, fair enough.